runtime/src/accounts_index.rs

```diff
@@ -581,6 +622,36 @@ impl<T: 'static + Clone> AccountsIndex<T> {
         max_root
     }
 
+    pub fn update_secondary_indexes(
+        &self,
+        slot: Slot,
+        pubkey: &Pubkey,
+        account_owner: &Pubkey,
+        account_data: &[u8],
+    ) {
+        if *account_owner == inline_spl_token_v2_0::id() && account_data.len() >= PUBKEY_BYTES {
+            let mint_pubkey = Pubkey::new(&account_data[..PUBKEY_BYTES]);
+            {
+                let mint_account_keys =
+                    self.mint_index.index.get(&mint_pubkey).unwrap_or_else(|| {
+                        self.mint_index
+                            .index
+                            .entry(mint_pubkey)
+                            .or_insert(DashSet::new())
+                            .downgrade()
+                    });
+                mint_account_keys.insert(*pubkey);
+            }
+
+            // TODO: handle if pubkey switches mint within the same slot
```

Comment:
I don't think this is possible? In order to switch a mints, a token account would have to close (lamports zeroed) and be removed before being reinitialized with the new mint.

Reply:
Ah, this is possible with serial transactions in the same slot.